cast/src/main/java/com/ibm/wala/cast/ipa/callgraph/CrossLanguageCallGraph.java

```diff
@@ -62,37 +63,37 @@ public CrossLanguageCallGraph(
 
   private final Set<CGNode> languageRootNodes = HashSetFactory.make();
 
-  private final Map<Atom, IMethod> languageRoots = HashMapFactory.make();
+  private final Map<Atom, @NonNull AbstractRootMethod> languageRoots = HashMapFactory.make();
 
   public AbstractRootMethod getLanguageRoot(Atom language) {
-    if (!languageRoots.containsKey(language)) {
-      AbstractRootMethod languageRoot = roots.get(language, this);
+    return languageRoots.computeIfAbsent(
+        language,
+        absent -> {
```

Comment:
should the parameter be named `language`?  It is actually used

Reply:
We cannot name this lambda's parameter `language`, because that would shadow the existing `language` parameter to the `getLanguageRoot` method itself. Java doesn't allow that kind of shadowing.

I suppose we could call it `absentLanguage` or `key` or `newLanguage` if you prefer any of those. Or we could rename the `language` parameter to `getLanguageRoot`.

Reply:
Easily done! Thanks for raising an eyebrow when you saw confusing code.